src/Equation.jl

```diff
@@ -384,7 +385,7 @@ function Base.hash(tree::Node{T})::UInt where {T}
     if tree.degree == 0
         if tree.constant
             # tree.val used.
-            return hash((0, tree.val::T))
+            return hash((0, tree.val))
```

Comment:
Why was this deleted? This type assertion helps quite a bit for performance

Reply:
Ah, sorry! I experimented with using the name in the hash.